pymodbus/client/serial.py

```diff
@@ -141,6 +141,7 @@ def __init__(
 
         self.last_frame_end = None
 
+        assert isinstance(self.comm_params.baudrate, int)
```

Comment:
No assert in production code please. Apart from that this is a strange place to assert on an internal struct...I could understand (but not accept) if you did assert on a parameter.

Reply:
The intention here was to [narrow the type](https://mypy.readthedocs.io/en/stable/type_narrowing.html) to an `int`.

However, the better solution is to find where `mypy` does not understand the default value (which _is_ an `int`) will be used.  There is probably a small bug with parameters in subclasses.